go/pkg/project/checkout.go

```diff
@@ -2,56 +2,49 @@ package project
 
 import (
 	"fmt"
-	"path"
 	"path/filepath"
 
 	"github.com/replicate/replicate/go/pkg/console"
 	"github.com/replicate/replicate/go/pkg/errors"
 )
 
 func (p *Project) CheckoutCheckpoint(checkpoint *Checkpoint, experiment *Experiment, outputDir string, quiet bool) error {
-	experimentFilesExist := true
-	checkpointFilesExist := true
+	if checkpoint == nil {
```

Comment:
`checkpoint` shouldn't be nil here anymore, so this should probably live somewhere in cli/checkpoint.go.

Reply:
Looking through the logic in `cli/checkpoint.go`, I don't think that is the case. That's what I was saying when saying the semantics between CLI and Python were different, but perhaps I wasn't clear enough.

I'll add a todo to refactor. :)

Reply:
Maybe I'll make missing tarballs through errors too, otherwise Python will fail silently.

Reply:
This has changed since last you looked, I think by the time you hit this function `checkpoint` will always be non-nil.

Reply:
I could be forgetting some case though...